tracer/src/Datadog.Trace/Telemetry/TelemetrySettings.cs

```diff
@@ -0,0 +1,54 @@
+﻿// <copyright file="TelemetrySettings.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.Telemetry
+{
+    internal class TelemetrySettings
+    {
+        public TelemetrySettings(IConfigurationSource source, ImmutableTracerSettings tracerSettings)
+        {
+            TelemetryEnabled = source?.GetBool(ConfigurationKeys.Telemetry.Enabled) ??
+                               // default value
+                               true;
+
+            var requestedTelemetryUri = source?.GetString(ConfigurationKeys.Telemetry.Uri);
+
+            if (!string.IsNullOrEmpty(requestedTelemetryUri) && Uri.TryCreate(requestedTelemetryUri, UriKind.Absolute, out var telemetryUri))
+            {
+                // if we're using a custom telemetry Uri, then it could be an external intake and need the api key
+                TelemetryUrl = telemetryUri;
```

Comment:
nit: Should all of the Telemetry "URIs" instead be "URLs"? To be honest URI vs URL trips me up so I'm not sure which one is technically correct here 🙈

Reply:
Maybe 😄 I never know the difference tbh, it seems like pedantry 😉 But yeah, will update them to URI because why not

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2241/commits/0dae0ef16b3e2935df41069516ede0954324a4ca